File: packages/lector/test/fixtures/typescript-reference/history/v2/payment.ts
export interface PurchaseOrder {
	readonly id: string;
	readonly amountCents: number;
	readonly currency: "USD" | "EUR";
}

export interface Receipt {
	readonly transactionId: string;
}

export interface PaymentProcessor {
	process(order: PurchaseOrder): Promise<Receipt>;
}

File: packages/lector/test/fixtures/python-reference/history/v1/payment.py
from abc import ABC, abstractmethod


class Order:
    def __init__(self, amount: int) -> None:
        self.amount = amount


class Receipt:
    def __init__(self, order: Order, processed: bool) -> None:
        self.order = order
        self.processed = processed


class PaymentProcessor(ABC):
    @abstractmethod
    def process(self, order: Order) -> Receipt: ...
